packages/client/tests/functional/esbuild-transformer.js

```diff
@@ -0,0 +1,62 @@
+const fs = require('fs')
+
+const esbuild = require('esbuild')
+const getCacheKeyFunction = require('@jest/create-cache-key-function').default
+
+const cacheKeyFunction = getCacheKeyFunction([], ['v1'])
+
+function needsTranspilation(contents, filename) {
+  if (filename.endsWith('.ts') === true) {
+    return true
+  }
+
+  if (contents.includes('require(') === true) {
+    return false
+  }
+
+  if (contents.includes('module.exports') === true) {
+    return false
+  }
+
+  if (contents.includes('exports.') === true) {
+    return false
+  }
+
+  return false
+}
+
+const transformer = {
+  getCacheKey(contents, filename, ...rest) {
+    return cacheKeyFunction(filename, fs.statSync(filename).mtimeMs + '', ...rest)
+  },
+  process(_content, filename, { transformerConfig }) {
+    if (needsTranspilation(_content, filename) === true) {
+      return esbuild.transformSync(_content, {
+        sourcesContent: true,
+        minify: false,
+        sourcefile: filename,
+        loader: 'ts',
+        format: 'cjs',
+        platform: 'node',
+        target: 'ES2020',
+        keepNames: true,
+        logLevel: 'error',
+        sourcemap: true,
+        ...transformerConfig,
+      })
+    }
+
+    if (fs.existsSync(`${filename}.map`) === true) {
+      return {
+        code: _content,
+        map: fs.readFileSync(`${filename}.map`, 'utf8'),
+      }
+    }
```

Comment:
```suggestion
    // We try to read the file
    // if it doesn't exist we ignore the error
    // it is supposed to be faster than checking if it exists first and then read it
    let mapFile
    try {
        return {
           code: _content,
           map: fs.readFileSync(`${filename}.map`, 'utf8'),
      	}
      } catch (e) {
        if (e.code !== 'ENOENT') throw e
      }
```
I don't know how many files we are talking about, this should be faster since it would be one read operation instead of 2. No idea how much impact it would have here without testing it 😄